File: app/src/main/java/com/tkdev/ironmanstash/infinity_stones/fragments/details/SingleStonePagerAdapter.java
package com.tkdev.ironmanstash.infinity_stones.fragments.details;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import java.util.List;

public class SingleStonePagerAdapter extends FragmentStatePagerAdapter {


    private List<SingleStone> singleStoneList;

    public SingleStonePagerAdapter(FragmentManager fm, List<SingleStone> singleStoneList) {
        super(fm);
        this.singleStoneList = singleStoneList;
    }


    @Override
    public Fragment getItem(int posInList) {

        SingleStone stone = singleStoneList.get(posInList);

        return SingleStoneFragmentDetail.newInstance(
                stone.getName(),
                stone.getColor(),
                stone.getQuest(),
                stone.getPassword()
        );
    }

    @Override
    public int getCount() {
        return singleStoneList.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return String.valueOf(position + 1);

    }
}

File: app/src/main/java/com/tkdev/ironmanstash/infinity_stones/fragments/details/SingleStonePagerAdapter.kt
package com.tkdev.ironmanstash.infinity_stones.fragments.details

import SingleStoneFragmentDetail
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class SingleStonePagerAdapter(fm: FragmentManager, private val singleStoneList: List<SingleStone>) : FragmentStatePagerAdapter(fm) {
    override fun getItem(posInList: Int): Fragment {
        val stone = singleStoneList[posInList]
        return SingleStoneFragmentDetail.newInstance(
                stone.name,
                stone.color,
                stone.quest,
                stone.password
        )
    }

    override fun getCount(): Int {
        return singleStoneList.size
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return (position + 1).toString()
    }

}